tests/test_cli_compile.py

```diff
@@ -662,7 +662,7 @@ def test_multiple_input_files_without_output_file(runner):
         (
             "--annotate",
             "small-fake-a==0.1         "
-            "# via -c constraints.txt (line 1), small-fake-with-deps\n",
+            "# via -c constraints.txt, small-fake-with-deps\n",
```

Comment:
It's possible to have duplicate constraints in one file, in that case the line number would be useful for disambiguation

Reply:
I mean, how useful? Wouldn't it be better to warn/error about duplicate constraints in a file, since they can always be combined into one line with commas?

Reply:
I'd be fine with something that just flagged that there are duplicates from a file, not necessarily saying which lines.  Also, another drawback I've discovered with the new output is that it makes merge conflicts much more likely; almost any two distinct changes to the input requirements or constraints will change many of the line numbers to different values.

Reply:
Ok maybe it's best to leave duplicates for another PR. The line number diffnoise is too much

Reply:
Yeah I'm merging more requirements changes today and the diffnoise is slowing down review quite a bit 😱

Made #1077 for warning on duplicated constraints.